components/formats-bsd/src/loci/formats/out/TiffWriter.java

```diff
@@ -214,19 +226,53 @@ public void saveBytes(int no, byte[] buf, IFD ifd, int x, int y, int w, int h)
     int type = FormatTools.pixelTypeFromString(
         retrieve.getPixelsType(series).toString());
     int index = no;
-    // This operation is synchronized
-    synchronized (this) {
-      // This operation is synchronized against the TIFF saver.
-      synchronized (tiffSaver) {
-        index = prepareToWriteImage(no, buf, ifd, x, y, w, h);
-        if (index == -1) {
-          return;
+    int imageWidth = retrieve.getPixelsSizeX(series).getValue().intValue();
+    int imageHeight = retrieve.getPixelsSizeY(series).getValue().intValue();
+    if (tileSizeX < imageWidth || tileSizeY < imageHeight) {
+      ifd.put(new Integer(IFD.TILE_WIDTH), new Long(tileSizeX));
+      ifd.put(new Integer(IFD.TILE_LENGTH), new Long(tileSizeY));
+    }
+    if (tileSizeX < w || tileSizeY < h) {
```

Comment:
I'm unsure of the logic used to select "tile mode" here.  Maybe the tile size should be set to zero when tiling isn't in use.  This would be necessary to allow tile sizes greater than the image size?